frontend/packages/dev-console/src/components/helm/HelmReleaseUpgradePage.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+import * as _ from 'lodash';
+import { safeLoad } from 'js-yaml';
+import { coFetchJSON, coFetch } from '@console/internal/co-fetch';
+import { LoadingBox } from '@console/internal/components/utils';
+import { HelmRelease, HelmChartMetaData, HelmActionType } from './helm-types';
+import { getChartURL } from './helm-utils';
+import HelmActionPage from './HelmActionPage';
+
+type HelmReleaseUpgradePageProps = RouteComponentProps<{ ns: string; releaseName: string }>;
+
+export const HelmReleaseUpgradePage: React.FC<HelmReleaseUpgradePageProps> = (props) => {
+  const {
+    match: {
+      params: { ns: namespace, releaseName },
+    },
+  } = props;
+
+  const [chartValues, setChartValues] = React.useState({});
+  const [chartName, setChartName] = React.useState('');
+  const [helmChartData, setHelmChartData] = React.useState<HelmChartMetaData[]>([]);
+  const [chartValuesLoaded, setChartValuesLoaded] = React.useState(false);
+  const [chartVersions, setChartVersions] = React.useState({});
+  const [activeChartVersion, setActiveChartVersion] = React.useState<string>('');
+
+  React.useEffect(() => {
+    const getChartDetails = () => {
+      coFetchJSON(`/api/helm/release?ns=${namespace}&release_name=${releaseName}`)
+        .then((res) => {
+          setActiveChartVersion(res.chart.metadata.version);
+          setChartName(res.chart.metadata.name);
+          setChartValues(res.chart.values);
+          setChartValuesLoaded(true);
+        })
+        .catch((err) => {
+          // eslint-disable-next-line no-console
+          console.error(err);
+        });
+    };
+
+    getChartDetails();
+  }, [namespace, releaseName]);
+
+  React.useEffect(() => {
+    const getChartVersions = () => {
+      let json: HelmRelease;
+
+      coFetch('/api/helm/charts/index.yaml')
+        .then(async (response) => {
+          const yaml = await response.text();
+          json = safeLoad(yaml);
+          const versions = _.reduce(
+            _.get(json, ['entries', chartName]),
+            (obj, chart) => {
+              obj[chart.version] = chart.version;
+              return obj;
+            },
+            {},
+          );
+          setHelmChartData(_.get(json, ['entries', chartName]));
+          setChartVersions(versions);
+        })
+        .catch((err) => {
+          // eslint-disable-next-line no-console
+          console.error(err);
+        });
+    };
+    getChartVersions();
+  }, [chartName]);
+
+  const onVersionChange = (value: string) => {
```

Comment:
Why do you need this logic here in this component. This could easily go into the dropdown component for chart version.

Reply:
I have kept it in the HelmInstallUpgradePage, as there I am fetching the entries of a helm chart, so every time user updates the version from the dropdown, to be able to fetch the values.yaml of that selected version I must fetch the url for that version. please do let me know if you still have a different opinion

Reply:
Also, in handleSubmit the helm chart entries is being used to fetch the chart url in case user wants to update the chart version